src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
    */
-  formatValue = (value) => {
-    if (this.isNaN(value)) {
-      return value;
-    }
+  const formatValue = useCallback(
+    (valueToFormat) => {
+      if (isNaN(valueToFormat)) {
+        return valueToFormat;
+      }
 
-    /* Guards against any white-space only strings like "   " being 
+      /* Guards against any white-space only strings like "   " being 
        mishandled and returned as `NaN` for visibleValue */
-    if (value === "" || value.match(/\s+/g)) {
-      return value;
-    }
+      if (valueToFormat === "" || valueToFormat.match(/\s+/g)) {
+        return valueToFormat;
+      }
 
-    const separator = this.getSeparator("decimal");
-    const [integer, remainder] = value.split(".");
+      const separator = getSeparator("decimal");
+      const [integer, remainder] = valueToFormat.split(".");
 
-    const formattedInteger = Intl.NumberFormat(
-      this.props.locale || this.context.locale(),
-      {
+      const formattedInteger = Intl.NumberFormat(locale || l.locale(), {
         maximumFractionDigits: 0,
+      }).format(integer);
+
+      let formattedNumber = formattedInteger;
+      if (remainder && remainder.length > precision) {
+        formattedNumber += `${separator + remainder}`;
+      } else if (remainder && remainder.length <= precision) {
+        formattedNumber += `${
+          separator + remainder + "0".repeat(precision - remainder.length)
+        }`;
+      } else {
+        formattedNumber += `${
+          precision ? separator + "0".repeat(precision) : ""
+        }`;
       }
-    ).format(integer);
-
-    let formattedNumber = formattedInteger;
-    if (remainder && remainder.length > this.props.precision) {
-      formattedNumber += `${separator + remainder}`;
-    } else if (remainder && remainder.length <= this.props.precision) {
-      formattedNumber += `${
-        separator +
-        remainder +
-        "0".repeat(this.props.precision - remainder.length)
-      }`;
-    } else {
-      formattedNumber += `${
-        this.props.precision ? separator + "0".repeat(this.props.precision) : ""
-      }`;
+      return formattedNumber;
+    },
+    [getSeparator, isNaN, l, locale, precision]
+  );
+
+  // Return previous value before state is changed. Used to compare prevState and newState.
+  function usePrevious(arg) {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = arg;
+    });
+    return ref.current;
+  }
+
+  /**
+   * Determine if the precision value has changed from the previous ref value for precision
+   */
+  const prevPrecisionValue = usePrevious(precision);
+
+  useEffect(() => {
+    if (prevPrecisionValue && prevPrecisionValue !== precision) {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Decimal `precision` prop has changed value. Changing the Decimal `precision` prop has no effect."
+      );
     }
-    return formattedNumber;
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Lets remove this comment and add the deps it tells you to

Reply:
I'll get this comment removed. Good spot.